alerting/src/test/kotlin/org/opensearch/alerting/util/clusterMetricsMonitorHelpers/CatIndicesWrappersIT.kt

```diff
@@ -0,0 +1,173 @@
+        assertNull(requestWrapper.indicesStatsRequest.indices())
+    }
+
+    fun `test CatIndicesRequestWrapper validate empty pathParams`() {
+        // GIVEN
+        val pathParams = ""
+
+        // WHEN
+        val requestWrapper = CatIndicesRequestWrapper(pathParams = pathParams)
+
+        // THEN
+        assertNull(requestWrapper.clusterHealthRequest.indices())
+        assertEquals(Strings.EMPTY_ARRAY, requestWrapper.clusterStateRequest.indices())
+        assertEquals(Strings.EMPTY_ARRAY, requestWrapper.indexSettingsRequest.indices())
+        assertNull(requestWrapper.indicesStatsRequest.indices())
+    }
+
+    fun `test CatIndicesRequestWrapper validate invalid pathParams`() {
+        // GIVEN
+        val pathParams = "_index1,index^2"
+
+        // WHEN & THEN
+        assertThrows(IllegalArgumentException::class.java) { CatIndicesRequestWrapper(pathParams = pathParams) }
+    }
+
+    suspend fun `test CatIndicesResponseWrapper returns with only indices in pathParams`() {
+        // GIVEN
+        val testIndices = (1..5).map {
+            "test-index${randomAlphaOfLength(10).lowercase()}" to randomIntBetween(1, 10)
+        }.toMap()
+
+        testIndices.forEach { (indexName, docCount) ->
+            repeat(docCount) {
+                val docId = (it + 1).toString()
+                val docMessage = """
+                    { 
+                        "message": "$indexName doc num $docId" 
+                    }
+                """.trimIndent()
+                indexDoc(indexName, docId, docMessage)
+            }
+        }
+
+        /*
+        Creating a subset of indices to use for the pathParams to test that all indices on the cluster ARE NOT returned.
+        */
+        val pathParamsIndices = testIndices.keys.toList().subList(1, testIndices.size - 1)
+        val pathParams = pathParamsIndices.joinToString(",")
+        val input = randomClusterMetricsInput(path = path, pathParams = pathParams)
+
+        // WHEN
+        val responseMap = (executeTransportAction(input, client())).toMap()
+
+        // THEN
+        val shards = responseMap[WRAPPER_FIELD] as List<HashMap<String, String>>
+        val returnedIndices =
+            shards.map { (it[CatIndicesResponseWrapper.IndexInfo.INDEX_FIELD] as String) to it }.toMap()
+
+        assertEquals(pathParamsIndices.size, returnedIndices.keys.size)
+        testIndices.forEach { (indexName, docCount) ->
+            if (pathParamsIndices.contains(indexName)) {
+                assertEquals(
+                    indexName,
+                    returnedIndices[indexName]?.get(CatIndicesResponseWrapper.IndexInfo.INDEX_FIELD) as String
+                )
+                assertEquals(
+                    docCount.toString(),
+                    returnedIndices[indexName]?.get(CatIndicesResponseWrapper.IndexInfo.DOCS_COUNT_FIELD) as String
+                )
+            }
+        }
+    }
```

Comment:
Future follow up to have a real cat indices call and compare all the values and not some.